diff --git a/src/libext/tblite/build_tblite_cmake.sh b/src/libext/tblite/build_tblite_cmake.sh
index 41a720ab36..3787824d2a 100755
--- a/src/libext/tblite/build_tblite_cmake.sh
+++ b/src/libext/tblite/build_tblite_cmake.sh
@@ -82,6 +82,7 @@ if [[ -n ${FC} ]] &&   [[ ${FC} == ftn ]]; then
 fi
 
 FC_EXTRA=$(${NWCHEM_TOP}/src/config/strip_compiler.sh ${FC})
+echo FC_EXTRA is $FC_EXTRA
 
 #Intel MPI
 if [[  -z "$I_MPI_F90"   ]] ; then
@@ -196,7 +197,9 @@ fi
 if [[ ${FC} == nvfortran ]] || [[ ${FC} == pgf90 ]]; then
   Fortran_FLAGS+="-Mbackslash -fast -tp host"
 fi
-
+if [ ! -z "${CONDA_BUILD_SYSROOT}" ]; then
+ Fortran_FLAGS+="-Wl,-rpath,${PREFIX}/lib -L${PREFIX}/lib"
+fi
 if [[ -z "$USE_OPENMP" ]]; then
   DOOPENMP=OFF
 else
@@ -219,5 +224,15 @@ fi
 $CMAKE --install _build
 
 cd ..
-strip --strip-debug ../lib/libtblite.a
+if [ ! -z "${CONDA_TOOLCHAIN_HOST}" ]; then
+    arch=$(echo $CONDA_TOOLCHAIN_HOST | cut -d - -f 1)
+else
+    arch=$(uname -m)
+fi
+if [[ "$arch" == "x86_64" ]]; then
+    strip --strip-debug ../lib/libtblite.a
+else
+    file ../lib/libtblite.a
+    file `find . -name "*gfn2*.o"`
+fi
 ln -sf  ../lib/libtblite.a  ../lib/libnwc_tblite.a
